### PR TITLE
fixes #53, Add support to interact with private docker registry

### LINKED ISSUE #53: Feature: Add support for private docker repository
Add support to ECS cluster to fetch docker images from a private repository which require authentication.

**Acceptance Criteria:**
* ECS should be able to fetch docker image from a private repository using authentication
* ECS should be able to fetch docker image from a public repository which does not require authentication

### PR DESCRIPTION
**Pull request by pradeepbhadani**

### CODE REVIEW

File: version.tf [hunk 1]
```diff
@@ -6,8 +6,8 @@
 
 terraform {
   required_version = "~> 0.11.1"
-}
 
-provider "aws" {
-  version = "~> 1.60.0"
+  required_providers = {
```
Comment (massdosage) [inline version.tf]: Should we mention this in the CHANGELOG or is it the equivalent to `provider` which we had before so isn't a visible change to users?

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) a
 
 ### Added
 - Improved error handling in scripts/endpoint_dns_name.sh - see [#17](https://github.com/ExpediaInc/apiary-federation/issues/17).
+- Support for docker private registry.
```
Comment (massdosage) [inline CHANGELOG.md]: docker -> Docker

File: version.tf (hunk 1 above)
Comment (pradeepbhadani) [reply version.tf]: this is equivalent to previous change so no change from user side.

File: CHANGELOG.md (hunk 2 above)
Comment (pradeepbhadani) [reply CHANGELOG.md]: updated

**Pull request merged**